tck/src/main/java/ee/jakarta/tck/data/standalone/entity/EntityTests.java

```diff
@@ -614,7 +627,17 @@ public void testFindFirst() {
     @Assertion(id = "133",
                strategy = "Use a repository method with findFirst3By that returns the first 3 results.")
     public void testFindFirst3() {
-        AsciiCharacter[] found = characters.findFirst3ByNumericValueGreaterThanEqualAndHexadecimalEndsWith(40, "4", Sort.asc("numericValue"));
+        AsciiCharacter[] found;
+        
+        try {
+        found = characters.findFirst3ByNumericValueGreaterThanEqualAndHexadecimalEndsWith(40, "4", Sort.asc("numericValue"));
```

Comment:
When we talk about query flexibility, Documents, and graphs allow similar queries to those in the relational world.

In this case, none of NoSQL might not support those.

![Screenshot 2024-06-23 at 07 03 54](https://github.com/jakartaee/data/assets/863011/0206c3e2-a3f4-43b6-891a-7484aa9cb1f0)


There is a huge universe of NoSQL databases, but doing a ballparking of those:

***Key-value**: It will only support key-based operations using equals and in. Anything else, ignore.
* **Wild-column** or column-family: Most keywords might work, but it needs to check the field itself. 


| Keyword         | Graph                                | Document                              | Column                                     | Key-value |
|-----------------|--------------------------------------|---------------------------------------|--------------------------------------------|-----------|
| `Between`       | Yes                                  | Not guaranteed (varies with vendor)   | Yes, but using ID or index field (Cassandra) | Nope      |
| `Contains`      | Not guaranteed (varies with vendor)  | Not guaranteed (varies with vendor)   | Not guaranteed (varies with vendor)        | Nope      |
| `EndsWith`      | Not guaranteed (varies with vendor)  | Not guaranteed (varies with vendor)   | Not guaranteed (varies with vendor)        | Nope      |
| `LessThan`      | Yes                                  | Yes                                   | Yes                                        | Nope      |
| `GreaterThan`   | Yes                                  | Yes                                   | Yes                                        | Nope      |
| `LessThanEqual` | Yes                                  | Yes                                   | Yes                                        | Nope      |
| `GreaterThanEqual` | Yes                               | Yes                                   | Yes                                        | Nope      |
| `Like`          | Not guaranteed (varies with vendor)  | Not guaranteed (varies with vendor)   | Not guaranteed (varies with vendor)        | Nope      |
| `IgnoreCase`    | Not guaranteed (varies with vendor)  | Not guaranteed (varies with vendor)   | Not guaranteed (varies with vendor)        | Nope      |
| `In`            | Yes                                  | Yes                                   | Yes, but using ID or index field (Cassandra) | Yes (ID or key only) |
| `Null`          | Not guaranteed (varies with vendor)  | Not guaranteed (varies with vendor)   | Not guaranteed (varies with vendor)        | Nope      |
| `StartsWith`    | Not guaranteed (varies with vendor)  | Not guaranteed (varies with vendor)   | Not guaranteed (varies with vendor)        | Nope (there is Redis where you can do regex in the key, but it only by the key: [Redis KEYS](https://redis.io/docs/latest/commands/keys/)) |
| `True`          | Yes                                  | Yes                                   | Yes, but using ID or index field (Cassandra) | Nope      |
| `False`         | Yes                                  | Yes                                   | Yes, but using ID or index field (Cassandra) | Nope      |
| `And`         | Yes                                  | Yes                                   | Yes, but using ID or index field (Cassandra) | Nope      |
| `Or`         | Yes                                  | Yes              | Nope | Nope      |
| Sort Operation         | Yes                                  | Yes              | Not guaranteed (varies with vendor and the field) | Nope      |

Reply:
⚠️ IMHO: key-value database should only support the insert, update, delete and save operations.
The queries such as Jakarta Data Query Language and method by query those does not make sense to support it.